subscription.bicep

```
@@ -0,0 +1,482 @@
+                        equals: 'Microsoft.Compute/virtualMachineScaleSets'
+                    }
+                    {
+                        field: 'Microsoft.Compute/virtualMachineScaleSets/virtualMachineProfile.networkProfile.networkInterfaceConfigurations[*].ipConfigurations[*].publicIPAddressConfiguration.name'
+                        like: '*'
+                    }
+                ]
+            }
+            then: {
+                effect: 'deny'
+            }
+        }
+    }
+}
+
+@description('Hubs policies deployment')
+module hubsPoliciesDeployment 'modules/hubsPoliciesDeployment.bicep' = {
+    name: 'Apply-${rgHubs.name}-Policies'
+    scope: rgHubs
+}
+
+@description('Spokes policies deployment')
+module spokesPoliciesDeployment 'modules/spokesPoliciesDeployment.bicep' = {
+    name: 'Apply-${rgSpokes.name}-Policies'
+    scope: rgSpokes
+}
+
+@description('Network watcher\'s policies deployment')
+module networkWatchersPoliciesDeployment 'modules/networkWatchersPoliciesDeployment.bicep' = {
+    name: 'Apply-${rgNetworkWatchers.name}-Policies'
+    scope: rgNetworkWatchers
+}
+
+@description('Workload\'s policies deployment')
+module workloadPoliciesDeployment 'modules/workloadPoliciesDeployment.bicep' = {
+    name: 'Apply-${rgbu0001a0005.name}-Policies'
+    scope: rgbu0001a0005
+}
+
+@description('Ensures that Microsoft Defender for Kuberentes Service, Container Service, and Key Vault are enabled. - Policy Assignment')
+module defenderPolicyDeployment 'modules/subscriptionPolicyAssignment.bicep' = {
+    name: 'Apply-EnableDefender-Policy'
+    scope: subscription()
+    params: {
+        location: deploymentResourceRegion
+        policyAssignmentIdentity: {
+            type: 'SystemAssigned'
+        }
+        polcyAssignmentMetadata: {
+            version: '1.0.0'
+            category: 'Microsoft Defender for Cloud'
+        }
+        policyDefinitionName: psdEnableDefender.name
+        policyAssignmentDescription: 'Ensures that Microsoft Defender for Kuberentes Service, Container Service, and Key Vault are enabled.'
+        enforcementMode: enforceAzureDefenderAutoDeployPolicies ? 'Default' : 'DoNotEnforce'
+        notScopes: []
+    }
+}
+
+@description('Enable Microsoft Defender Standard for Key Vault. Requires Owner or Security Admin role.')
+resource enableKeyVaultspricing 'Microsoft.Security/pricings@2018-06-01' = if (enableMicrosoftDefenderForCloud) {
+    name: 'KeyVaults'
+    properties: {
+        pricingTier: 'Standard'
+    }
+}
+
+@description('Enable Microsoft Defender Standard for Container Registry. Requires Owner or Security Admin role.')
+resource enableContainerRegistry 'Microsoft.Security/pricings@2018-06-01' = if (enableMicrosoftDefenderForCloud) {
+    name: 'ContainerRegistry'
+    properties: {
+        pricingTier: 'Standard'
+    }
+}
+
+@description('Enable Microsoft Defender Standard for Kubernetes Service. Requires Owner or Security Admin role.')

```

Just so we don't get any questions.  It's known :)

```suggestion
@description('Enable Microsoft Defender for Kubernetes Service. Deprecated, please move to Defender for Containers. Requires Owner or Security Admin role.')
```